frontend-react/src/components/Table/TableFilters.tsx

```diff
@@ -50,8 +48,10 @@ function TableFilters({
     const [rangeFrom, setRangeFrom] = useState<string>(FALLBACK_FROM);
     const [rangeTo, setRangeTo] = useState<string>(FALLBACK_TO);
 
-    let from = new Date(rangeFrom).toISOString();
-    let to = inclusiveDateString(rangeTo);
+    let from = !isNaN(Date.parse(rangeFrom))
+        ? new Date(rangeFrom).toISOString()
+        : "";
+    let to = !isNaN(Date.parse(rangeTo)) ? inclusiveDateString(rangeTo) : "";
```

Comment:
We should just put the raw values `rangeFrom` into `updateRange` function. declaring them as variables and using that to typecheck them isn't ideal, nor is using them outside of a react hook like `useMemo` to make sure it has the proper dependencies.

My recommendation:

Let's do that `Date.parse(input)` type check within the onChange function:

```
onChange: (val?: string) => {
                            if (val && Date.parse(val)) {
                                setRangeTo(val);
                            }
                        },
```

`Date.parse()` returns `NaN` when it's a non-valid date which is falsy.

Reply:
Agreed -- we should be validating/parsing prior to committing to state rather than doing it on the subsequent render 🙏